pkg/gofr/datasource/kv-store/badger/badger.go

```diff
@@ -0,0 +1,103 @@
+package badger
+
+import (
+	"context"
+	"github.com/dgraph-io/badger/v4"
+)
+
+type Configs struct {
+	DirPath string
+}
+
+type client struct {
+	db      *badger.DB
+	configs Configs
+	logger  Logger
+	metrics Metrics
+}
+
+func New(configs Configs) *client {
+	return &client{configs: configs}
+}
+
+// UseLogger sets the logger for the MongoDB client which asserts the Logger interface.
+func (c *client) UseLogger(logger interface{}) {
```

Comment:
We are far from Go 1.18, you can use any


```suggestion
func (c *client) UseLogger(logger any) {
```